src/main.py

```diff
@@ -1,125 +1,246 @@
-        if client.user != reaction.message.author:
-            return
+    # Send posts from listening channels to verification channel
+    if message.channel.id in ART_CHANS:
+        await verify_art(message=message)
+
+@bot.tree.context_menu(name=strings.get("app_name_publish"), guilds=[guild.id for guild in bot.guilds])
+async def command_publish(interaction: discord.Interaction, message: discord.Message):
+    """
+    Reposts a message from a self-curated or verification channel as a formatted embed to a showcase or gallery channel,
+    respectively, depending on the channel the original message was posted in.
+    Posts in showcase channels may also be removed with this command.
+    """
+    reply = None
+    success = False      # Reactions are added to messages based on success or failure
+    fail_quietly = True  # Failure reactions are hidden by default
+
+    # We avoid duplicate publish actions by checking for reactions
+    posted = False
+    for reaction in message.reactions:
+        posted = bot.user in [user async for user in reaction.users()]
+        if posted:
+            break
+
+    # Interactions on verification channel posts by this bot
+    if message.channel.id == VERIFY_CHAN:
+
+        # Verification errors are shown to all to avoid repeat attempts
+        fail_quietly = False
+
+        # Ignore interactions from users without any of the required roles
+        if not check_roles(interaction.user, VERIFY_ROLES):
+            reply = format_roles_error(strings.get("commands_error_roles"), VERIFY_ROLES)
+
+        # Ignore messages that have been handled previously
+        elif posted:
+            reply = strings.get("publish_error_posted")
+            fail_quietly = True
+
+        # Ignore reactions to invalid posts or posts not by this client in verification channel
+        elif bot.user != message.author or not await publish_art(message=message):
+            reply = strings.get("publish_error_user")
 
         # Publish verified posts to gallery channel
-        await publish_art(message=reaction.message)
+        else:
+            reply = strings.get("publish_response_verified").format(f"<#{GALLERY_CHAN}>")
+            success = True
+
+    # User interactions on posts in self-curated channel
+    elif message.channel.id in MOD_CHANS:
+
+        # Ignore interactions from users without any of the required roles
+        if not check_roles(interaction.user, SHOWCASE_ROLES):
+            reply = format_roles_error(strings.get("commands_error_roles"), SHOWCASE_ROLES)
 
-    # User reactions to posts in self-curated channel
-    elif reaction.message.channel.id in MOD_CHANS:
-        # Ignore reactions from users other than the message author
-        if user != reaction.message.author:
-            return
+        # Ignore interactions from users other than the message author
+        elif interaction.user != message.author or not isinstance(interaction.user, discord.Member):
```

Comment:
Do we need to check if `interaction.user` is a member? afaik the only way this could be false is in DMs

Reply:
I forget why this came up, it mightve been related to role checks or user info fetching to quiet down the type hint warnings from pycharm, I'll look at it again when I'm home

Reply:
No warnings on removing it, change pushed